pkg/3scale/amp/component/system.go

```diff
@@ -250,6 +255,27 @@ func (system *System) buildSystemBaseEnv() []v1.EnvVar {
 		)
 	}
 
+	if system.Options.STS && system.Options.S3STSEnabled != nil {
+		result = append(result,
+			helper.EnvVarFromConfigMap("FILE_UPLOAD_STORAGE", "system-environment", "FILE_UPLOAD_STORAGE"),
+			helper.EnvVarFromSecret(AwsRoleArn, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsRoleArn),
+			helper.EnvVarFromSecret(AwsWebIdentityTokenFile, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsWebIdentityTokenFile),
+			helper.EnvVarFromSecret(AwsBucket, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsBucket),
+			helper.EnvVarFromSecret(AwsRegion, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsRegion),
+			helper.EnvVarFromSecretOptional(AwsProtocol, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsProtocol),
+			helper.EnvVarFromSecretOptional(AwsHostname, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsHostname),
+			helper.EnvVarFromSecretOptional(AwsPathStyle, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsPathStyle),
+
+			helper.EnvVarFromSecret(AwsRoleArn, system.Options.S3STSEnabled.ConfigurationSecretName, AwsRoleArn),
```

Comment:
duplicated?

Reply:
@eguzki , thank you for comment, this is what I wanted to ask -  https://github.com/3scale/3scale-operator/blob/c87086c823cad919d66016ab19e65010a368d707/pkg/3scale/amp/component/system.go#L258-L275    
Maybe we don't need S3STSEnabled (Struct)? Actually, seems I'm not using it. But using system.Options.STS (bool)